### PR TITLE
Simplify Phoenix error template

### PR DESCRIPTION
**Pull request by vinibrsl**

### CODE REVIEW

File: lib/plausible_web/templates/layout/error.html.eex [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
```
Comment (ukutaht) [inline lib/plausible_web/templates/layout/error.html.eex:1]: AFAICT within this template we're rendering two inner templates:
* lib/plausible_web/templates/error/error.html.eex
* lib/plausible_web/templates/error/server_error.html.eex

The first one also includes the whole HTML layout code. I think that should be adjusted to only contain what's needed in the `<%= @inner_content %>` of this layout.

File: lib/plausible_web/controllers/helpers.ex [hunk 2]
```diff
@@ -6,13 +6,16 @@ defmodule PlausibleWeb.ControllerHelpers do
     conn
     |> put_status(status)
     |> put_view(PlausibleWeb.ErrorView)
-    |> render("#{status}.html", layout: false, message: message)
+    |> render("#{status}.html", message: message, layout: error_layout())
```
Comment (vinibrsl) [inline lib/plausible_web/controllers/helpers.ex:9]: This module renders errors without the need of exceptions, for example when a controller wants to explicitly render a 404. Phoenix only automatically sets the layout when there is an exception, and for this case I needed to set the layout "manually".

File: lib/plausible_web/templates/layout/error.html.eex (hunk 1 above)
Comment (vinibrsl) [reply lib/plausible_web/templates/layout/error.html.eex]: You're right! I pushed a change removing the HTML boilerplate as we're using a layout now.

**Comment by vinibrsl** [comment]
I still need to fix one test that's failing.

File: lib/plausible_web/controllers/helpers.ex [hunk 3]
```diff
@@ -6,13 +6,16 @@ defmodule PlausibleWeb.ControllerHelpers do
...
   def render_error(conn, status) do
     conn
     |> put_status(status)
     |> put_view(PlausibleWeb.ErrorView)
-    |> render("#{status}.html", layout: false)
+    |> render("#{status}.html", layout: error_layout())
   end
+
+  defp error_layout,
+    do: Application.get_env(:plausible, PlausibleWeb.Endpoint)[:render_errors][:layout]
```
Comment (vinibrsl) [inline lib/plausible_web/controllers/helpers.ex:20]: This module renders errors without the need of exceptions, for example when a controller wants to explicitly render a 404. Phoenix automatically sets the layout only when there is an exception, and for this case I needed to set the layout "manually".

**Review by ukutaht — APPROVED** [review]

**Pull request merged**